lib/portage/sync/modules/git/git.py

```diff
@@ -117,6 +117,22 @@ def new(self, **kwargs):
 
         return (os.EX_OK, True)
 
+    def _gen_ceiling_string(self, path):
```

Comment:
Why can't we set GIT_CEILING_DIRECTORIES="${PARENT_OF_REPO}", e.g. /var/db/repos if my repo is /var/db/repos/gentoo?

If /var/db/repos/gentoo doesn't have a git repository, it might move up into /var/db/repos, but the environment variable forbids it. git doesn't search laterally as far as I know (imagine /var/db/repos/local with a git repository). Is that what you're aiming to prevent here?

After writing this out, I guess this might be needed with symlinks?

Reply:
The git manpage says, about the environment variable, "If set, it is a list of directories that Git should not chdir up into while looking for a repository". Does that imply it definitely won't chdir into `/var/db/` if `/var/db/repos/` is in the CEILING_DIRECTORIES? I can read it both ways. Maybe I'm just being overly cautious or misunderstanding the manpage, though.

Reply:
Ah! I was obsessing over how to interpret it to allow lateral movement, but allowing non-linear jumps makes more sense as something which an obnoxious implementation could do. Could you put that in a comment?